### PR TITLE
Sorted (grouped) results

### PR DESCRIPTION
**Pull request by lschatzkin**
A basic sort function to group the results by category

### CODE REVIEW

File: static/results.js [hunk 1]
```diff
@@ -13,87 +21,87 @@ $(document).ready(function(){
...
       if ($(this).html().includes('Device Memory')) {
         var filename = '/static/results-text/device-memory.txt';
-        var selfname = $(this);
+        var selfname = $(this).prepend('<span>4hardware <span>');
         getText(filename, selfname);
       }
       if ($(this).html().includes('DNT Header')) {
         var filename = '/static/results-text/dnt-header.txt';
-        var selfname = $(this);
+                var selfname = $(this).prepend('<span>1header <span>');
```
Comment (Hainish) [inline static/results.js:54]: I think this indentation is a mistype?

File: static/results.js [hunk 2]
```diff
@@ -133,14 +141,21 @@ $(document).ready(function(){
...
-      });
+    // smaller phone, make table of contents (see results template)
+    $('select').selectmenu({
+      'change': function () {
+        var val = $( "#characteristic option:selected" ).val();
+        console.log('val='+val);
+        window.location.href = val;
+      }
+    });
+  // sort results by group
```
Comment (Hainish) [inline static/results.js:152]: These should be indented one more scope

File: static/results.js [hunk 3]
```diff
@@ -13,87 +21,87 @@ $(document).ready(function(){
...
       if ($(this).html().includes('User Agent')) {
         var filename = '/static/results-text/user-agent.txt';
-        var selfname = $(this);
+        var selfname = $(this).prepend('<span>1header <span>');
         getText(filename, selfname);
       }
       if ($(this).html().includes('HTTP_ACCEPT Headers')) {
         var filename = '/static/results-text/http-accept-headers.txt';
-        var selfname = $(this);
+                var selfname = $(this).prepend('<span>1header <span>');
```
Comment (Hainish) [inline static/results.js:29]: I think this indentation is a mistype?

File: static/results.js [hunk 4]
```diff
@@ -13,87 +21,87 @@ $(document).ready(function(){
       }
       if ($(this).html().includes('User Agent')) {
         var filename = '/static/results-text/user-agent.txt';
-        var selfname = $(this);
+        var selfname = $(this).prepend('<span>1header <span>');
```
Comment (Hainish) [inline static/results.js:24]: How is this used?  It seems like an antipattern to sort via the HTML inside a span.  Maybe we can use a data attribute, like `<span data-sort-group='1'>`

**Review by Hainish — CHANGES_REQUESTED** [review]

File: static/results.js (hunk 4 above)
Comment (lschatzkin) [reply static/results.js:24]: good suggestion, will do.

**Pull request merged**